toolbox/src/base/ToolboxSingleton.cpp

```diff
@@ -0,0 +1,124 @@
+#include "ToolboxSingleton.h"
+
+#include "Log.h"
+#include "RobotInterface.h"
+#include <string>
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/ModelIO/ModelLoader.h>
+#include <yarp/os/ResourceFinder.h>
+#include <yarp/os/Network.h>
+
+// TODO: remove nesting
+namespace wbt {
+
+    // CONSTRUCTOR / DESTRUCTOR
+    // ========================
+
+    ToolboxSingleton::ToolboxSingleton() {}
+    ToolboxSingleton::ToolboxSingleton(const ToolboxSingleton&) {}
+    ToolboxSingleton::~ToolboxSingleton() {}
+    ToolboxSingleton& ToolboxSingleton::operator=(const wbt::ToolboxSingleton &) { return *this; }
+
+    // UTILITIES
+    // =========
+
+    int ToolboxSingleton::numberOfDoFs(const std::string& confKey)
+    {
+        if (!isKeyValid(confKey))
+            return -1;
+        else
+            return m_interfaces[confKey]->getConfiguration().getNumberOfDoFs();
+    }
+
+    bool ToolboxSingleton::isKeyValid(const std::string& confKey)
+    {
+        if (m_interfaces.find(confKey) != m_interfaces.end()) {
+            if (m_interfaces[confKey])
+                return true;
+            else
+                return false;
+        }
+        else {
+            return false;
+        }
+    }
+
+    // GET METHODS
+    // ===========
+
+    ToolboxSingleton& ToolboxSingleton::sharedInstance()
+    {
+        static ToolboxSingleton instance;
+        return instance;
+    }
+
+    const Configuration& ToolboxSingleton::getConfiguration(const std::string& confKey)
+    {
+        return getRobotInterface(confKey)->getConfiguration();
+    }
+
+    const std::shared_ptr<RobotInterface> ToolboxSingleton::getRobotInterface(const std::string& confKey)
+    {
+        if (!isKeyValid(confKey)) {
+            return nullptr;
+        }
+
+        return m_interfaces[confKey];
+    }
+
+    const std::shared_ptr<iDynTree::KinDynComputations> ToolboxSingleton::model(const std::string& confKey)
+    {
+        if (!isKeyValid(confKey)) {
+            return nullptr;
+        }
+
+        return m_interfaces[confKey]->getKinDynComputations();
+    }
+
+    // TOOLBOXSINGLETON CONFIGURATION
+    // ==============================
+
+    bool ToolboxSingleton::configure(const std::string& confKey, const Configuration& config)
+    {
+        if (!config.isValid()) {
+            return false;
+        }
+
+        // Add the new Configuration object and override an existing key if it already exist.
+        // Note: Simulink doesn't flush memory unless Matlab is closed, and static objects stay in memory.
+        //       This may cause problems if the config block's mask is changed after the first compilation.
+        if (m_interfaces.find(confKey) == m_interfaces.end()) {
+            m_interfaces[confKey] = std::make_shared<RobotInterface>(config);
+            return m_interfaces[confKey].get();
```

Comment:
This comment applies also to other uses of `.get()`.
`shared_ptr` have an `operator bool`. This means that you can directly return `return m_interfaces[confKey]` to assess validity of the pointer.

Change this line and the other times you use `get()` as a boolean check

Reply:
The conversion from shared_ptr to bool is `explicit`. An alternative to `get()` would be `static_cast<bool>`. Using this latter version.